src/main/java/org/openrewrite/java/testing/junit4toassertj/ChangeAssertTrue.java

```diff
@@ -0,0 +1,124 @@
+package org.openrewrite.java.testing.junit4toassertj;
+
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.*;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.openrewrite.Formatting.EMPTY;
+import static org.openrewrite.Formatting.format;
+import static org.openrewrite.Tree.randomId;
+
+/**
+ * This is a refactoring visitor that will convert JUnit-style assertTrue() to assertJ asserts:
+ *
+ * assertTrue("This should be true", EXPRESSION) -> assert(EXPRESSION).with("This should be true").isTrue()
+ * assertTrue(EXPRESSION) -> assert(EXPRESSION).isTrue()
+ *
+ */
+public class ChangeAssertTrue extends JavaIsoRefactorVisitor {
+
+    private static final MethodMatcher assertTrueMatcher = new MethodMatcher("org.junit.Assert assertTrue(..)");
+
+    public ChangeAssertTrue() {
+        setCursoringOn();
+    }
+
+    @Override
+    public J.CompilationUnit visitCompilationUnit(J.CompilationUnit cu) {
+
+        maybeRemoveImport("org.junit.Assert");
+        return super.visitCompilationUnit(cu);
+    }
+
+    @Override
+    public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method) {
+        J.MethodInvocation original = super.visitMethodInvocation(method);
+        if (!assertTrueMatcher.matches(method)) {
+            return original;
+        }
+
+        //The expression that evaluates to true/false is either the first argument (if using the one-argument form)
+        //or the second argument (if using the two-argument form)
```

Comment:
Between JUnit4 and 5 they switched the position of the optional "message" parameter. In 4 the message always comes first. In 5 the message always comes last. I don't think we'd want to say "Oops you already ran our recipe to go from 4 to 5, so now you can't use the assertJ recipe which only supports 4". So I suspect we'd either want this to support both JUnit 4 and 5, or just 5. What do you think?

Reply:
Yeah, I definitely wanted to support both migration from JUnit 4 and JUnit 5. My initial thinking here was to put all of the JUnit 4 visitors in the junit4toassertj package and then have similar visitors for junit5toassertj. This would keep each visitor single purpose. Would you prefer that we have one visitor for each type and then have logic for both types of asserts?